src/ophyd_async/epics/areadetector/drivers/ad_base.py

```diff
@@ -18,11 +37,29 @@ def __init__(self, prefix: str, name: str = "") -> None:
         self.array_counter = ad_rw(int, prefix + "ArrayCounter")
         self.array_size_x = ad_r(int, prefix + "ArraySizeX")
         self.array_size_y = ad_r(int, prefix + "ArraySizeY")
+        self.detector_state = ad_r(DetectorState, prefix + "DetectorState")
```

Comment:
I think Tom wants the ADBase here to replicate that of Malcolm? (@coretl )

I also know that this will be useful for the tetramm to have, but currently although the PilatusDriver and AdAravisMakoDriver in dodal (needed for d11 for example, for p38) subclass ADBase, the TetrammDriver does not. So, this functionality wouldn't exist for the tetramm.

I'm not sure where the best place for this would be. I agree it's here. I think there's a larger discussion that needs to happen with @coretl to decide where it ought to live.

Reply:
After speaking with @coretl seems like this is the right place after all :)